beekeeper-core/src/test/java/com/expediagroup/beekeeper/core/repository/HousekeepingMetadataRepositoryTest.java

```diff
@@ -362,10 +357,69 @@ public void dryRunCountPartitionsForUnpartitionedTable() {
+    HousekeepingMetadata housekeepingEntity = createPartitionedEntityHousekeepingTable(SCHEDULED);
+    housekeepingMetadataRepository.save(housekeepingEntity);
+
+    housekeepingMetadataRepository.deleteScheduledOrFailedPartitionRecordsForTable(DATABASE_NAME, TABLE_NAME);
+    List<HousekeepingMetadata> remainingPaths = Lists.newArrayList(housekeepingMetadataRepository.findAll());
+    assertThat(remainingPaths.size()).isEqualTo(0);
+  }
+
+  @Test
+  @Transactional
+  public void deleteScheduledOrFailedPartitionRecordsForTableNothingToDelete() {
+    HousekeepingMetadata housekeepingEntity = createPartitionedEntityHousekeepingTable(DELETED);
+    housekeepingMetadataRepository.save(housekeepingEntity);
+
+    housekeepingMetadataRepository.deleteScheduledOrFailedPartitionRecordsForTable(DATABASE_NAME, TABLE_NAME);
+    List<HousekeepingMetadata> remainingPaths = Lists.newArrayList(housekeepingMetadataRepository.findAll());
+    assertThat(remainingPaths.size()).isEqualTo(1);
+  }
+
+  @Test
+  @Transactional
+  public void deleteScheduledOrFailedPartitionRecordsForTableMixedStatus() {
+    HousekeepingMetadata housekeepingEntity = createPartitionedEntityHousekeepingTable(SCHEDULED);
+    housekeepingMetadataRepository.save(housekeepingEntity);
+    HousekeepingMetadata housekeepingEntity1 = createPartitionedEntityHousekeepingTable(DELETED);
+    housekeepingMetadataRepository.save(housekeepingEntity1);
+    HousekeepingMetadata housekeepingEntity2 = createPartitionedEntityHousekeepingTable(FAILED);
+    housekeepingMetadataRepository.save(housekeepingEntity2);
+
+    housekeepingMetadataRepository.deleteScheduledOrFailedPartitionRecordsForTable(DATABASE_NAME, TABLE_NAME);
+    List<HousekeepingMetadata> remainingPaths = Lists.newArrayList(housekeepingMetadataRepository.findAll());
+    assertThat(remainingPaths.size()).isEqualTo(1);
+  }
+
+  @Test
+  public void findActiveTables() {
+    HousekeepingMetadata table1 = createEntityHouseKeepingTable("db", "tbl1", null);
+    housekeepingMetadataRepository.save(table1);
+    HousekeepingMetadata table2 = createEntityHouseKeepingTable("db", "tbl2", null);
+    housekeepingMetadataRepository.save(table2);
+    HousekeepingMetadata copy = createEntityHouseKeepingTable("db", "tbl2", null);
+    housekeepingMetadataRepository.save(copy);
+    HousekeepingMetadata table3 = createEntityHouseKeepingTable("db", "tbl3", null);
+    table3.setHousekeepingStatus(DISABLED);
+    housekeepingMetadataRepository.save(table3);
+    HousekeepingMetadata partition = createPartitionedEntityHousekeepingTable();
+    housekeepingMetadataRepository.save(partition);
+    
+    List<HousekeepingMetadata> result = housekeepingMetadataRepository.findActiveTables();
+    assertThat(result.size()).isEqualTo(3);
+    assertThat(result.get(0).getTableName()).isEqualTo("tbl1");
+    assertThat(result.get(1).getTableName()).isEqualTo("tbl2");
+    assertThat(result.get(2).getTableName()).isEqualTo("tbl2");
```

Comment:
this should return distinct db.tables records right, otherwise you might pull out a lot of rows.

Reply:
there should be only 1 record for each table which has the partitionName=null

Reply:
got it